File: src/wordle/WordleUserInterface.java
package wordle;

final public class WordleUserInterface {
    private final WordleRulesController wordleRulesController;

    public WordleUserInterface(final WordleRulesController wordleRulesController) {
        this.wordleRulesController = wordleRulesController;
    }

    public void run() {
        while(!wordleRulesController.isGameOver()) {
            renderCandidateSolutions(wordleRulesController.getCandidateSolutions());
            wordleRulesController.guess(guessAWord());
        }
        renderCandidateSolutions(wordleRulesController.getCandidateSolutions());
    }

    private Word guessAWord() {
        return null; // TODO: Ask for the user to actually guess a word
    }

    private void renderCandidateSolutions(final CandidateSolutions candidateSolutions) {
        // TODO: Do UI magic here - colors, styles, etc. based on each word's LetterStates
    }
}

File: src/wordle/WordleUserInterface.kt
package wordle

class WordleUserInterface(private val wordleRulesController: WordleRulesController) {
    fun run() {
        while (!wordleRulesController.isGameOver) {
            renderCandidateSolutions(wordleRulesController.candidateSolutions)
            wordleRulesController.guess(guessAWord())
        }
        renderCandidateSolutions(wordleRulesController.candidateSolutions)
    }

    private fun guessAWord(): CharArray {
        // TODO: Ask for the user to actually guess a word
        return CharArray(wordleRulesController.totalPossibleLetters) { '-'}
    }

    private fun renderCandidateSolutions(candidateSolutions: CandidateSolutions) {
        // TODO: Do UI magic here - colors, styles, etc. based on each letter's value & LetterState
    }
}